src/back/glsl/mod.rs

```diff
@@ -597,36 +605,12 @@ impl<'a, W: Write> Writer<'a, W> {
                 write!(self.out, "]")?
             }
             // glsl structs are written as just the struct name if it isn't a block
-            //
-            // If it's a block we need to write `block_name { members }` where `block_name` must be
-            // unique between blocks and structs so we add `_block_ID` where `ID` is a `IdGenerator`
-            // generated number so it's unique and `members` are the same as in a struct
             TypeInner::Struct { block, ref members } => {
                 // Get the struct name
                 let name = &self.names[&NameKey::Type(ty)];
 
                 if block {
```

Comment:
we should probably just match it here, i.e. `TypeInner::Struct { block: true, ref members }` instead of `if/else` inside